mobile-app/app/components/TransactionCardWalletTextInputV2.tsx

```diff
@@ -0,0 +1,219 @@
+import { MaterialIcons } from '@expo/vector-icons'
+
+type WalletTextInputProps = React.PropsWithChildren<TextInputProps> & IWalletTextInputProps
+export type InputType = 'default' | 'numeric'
+
+interface IWalletTextInputProps {
+  inputType: InputType
+  title?: string
+  titleTestID?: string
+  valid?: boolean
+  editable?: boolean
+  inlineText?: {
+    type: 'error' | 'helper'
+    text?: string | JSX.Element
+    style?: StyleProp<ViewStyle>
+  }
+  displayClearButton?: boolean
+  onClearButtonPress?: () => void
+  displayFocusStyle?: boolean
+  containerStyle?: string
+  inputContainerStyle?: StyleProp<ViewStyle>
+  onBlur?: () => void
+  hasBottomSheet?: boolean
+  inputFooter?: React.ReactElement
+  displayTickIcon?: boolean
+}
+
+export const TransactionCardWalletTextInputV2 = forwardRef<any, WalletTextInputProps>(function (props: WalletTextInputProps, ref: React.Ref<any>): JSX.Element {
+  // const [isFocus, setIsFocus] = useState(false)
+  const {
+    title,
+    titleTestID,
+    valid = true,
+    inlineText,
+    editable = true,
+    displayClearButton = false,
+    onClearButtonPress,
+    children,
+    containerStyle,
+    inputContainerStyle,
+    onBlur,
+    hasBottomSheet,
+    inputFooter,
+    displayTickIcon,
+    ...otherProps
+  } = props
+
+  const textInputComponents = {
+    ios: TextInputIOS,
+    default: TextInputDefault
+  }
+  const TextInput = Platform.OS === 'ios' && hasBottomSheet === true ? textInputComponents.ios : textInputComponents.default
+
+  const hasClearButton = (): boolean => {
+    return (displayClearButton) && (onClearButtonPress !== undefined)
+  }
+
+  return (
+    <ThemedViewV2
+      light={tailwind('bg-transparent')}
+      dark={tailwind('bg-transparent')}
+      style={tailwind(`${containerStyle ?? 'w-full flex-col'}`)}
+    >
+      {title !== undefined &&
+        (
+          <ThemedSectionTitleV2
+            testID={titleTestID}
+            text={title}
+          />
+        )}
+      <ThemedViewV2
+        light={tailwind('bg-mono-light-v2-00 border-mono-light-v2-00', { 'border-red-v2': !valid })}
+        dark={tailwind('bg-mono-dark-v2-00 border-mono-dark-v2-00', { 'border-red-v2': !valid })}
+        style={tailwind('flex-col w-full border-0.5 rounded-lg-v2')}
+      >
+        <View
+          style={[tailwind('flex-row items-center py-2 pl-5 pr-3 justify-between'), props.multiline === true && { minHeight: 54 }, inputContainerStyle]}
+        >
+          <TextInput
+            // onFocus={() => setIsFocus(true)}
+            onBlur={() => {
+              if (onBlur !== undefined) {
+                onBlur()
+              }
+              // setIsFocus(false)
+            }}
+            ref={ref}
+            editable={editable}
+            style={tailwind('font-normal-v2 flex-1 h-5')}
+            selectionColor={getColor('brand-v2-500')}
+            {...otherProps}
+          />
+          {displayTickIcon === true &&
+            <MaterialIcons
```

Comment:
```suggestion
            <ThemedIcon
```

Reply:
since the tick will always display green for now, do we still need to use `<ThemedIcon />`?